### PR TITLE
Add new BBE for WebHook Service using WebSub Service

### PR DESCRIPTION
**Pull request by ayeshLK**
Related to [#823](https://github.com/ballerina-platform/ballerina-standard-library/issues/823)

### CODE REVIEW

File: examples/index.json [hunk 1]
```diff
@@ -1041,6 +1041,18 @@
             }
         ]
     },
+    {
+        "title": "WebSub",
+        "column": 1,
+        "category": "Working with WebSub Protocol",
+        "samples": [
+            {
+                "name": "GitHub WebHook Service",
```
Comment (shafreenAnfar) [inline examples/index.json:1050]: Let's rename it as `Websub Subscriber Service`

File: examples/websub-webhook-sample/github_webhook_service.bal [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    target: ["https://api.github.com/hub", "https://github.com/<YOUR_ORGANIZATION>/<YOUR_REPOSITORY>/events/push.json"],
+    secret: "<YOUR_SECRET_KEY>",
+    callback: "<YOUR_CALLBACK_URL>",
+    httpConfig: {
+        auth: {
+            token: "<YOUR_AUTH_TOKEN>"
+        }
+    }
+}
+service /subscriber on websubEP {
```
Comment (shafreenAnfar) [inline examples/websub-webhook-sample/github_webhook_service.bal:151]: `service /subscriber on new (9090); `

Lets do the above

File: examples/websub-webhook-sample/github_webhook_service.bal [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    Repository repository;
+    json pusher;
+    User sender;
+|};
+
+// Current supported GitHub WebHook event types.
+public type GitHubEvent PingEvent|PushEvent;
+
+// The endpoint to which the webhook service is inbound.
+listener websub:Listener websubEP = new (9090);
```
Comment (shafreenAnfar) [inline examples/websub-webhook-sample/github_webhook_service.bal:138]: Remove this.

File: examples/websub-webhook-sample/github_webhook_service.bal [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        var payload = event.content.cloneWithType(GitHubEvent);
+        if(payload is PingEvent) {
+            io:println(string`Ping Event received for webhook [${payload.hook_id}]. Event sender [${payload.sender.id}]`);
+        } else if (payload is PushEvent) {
+            io:println(string`Push Event received for repository [${payload.repository.name}]. Last updated at ${payload.repository.updated_at}`);
+        } else {
+            io:println("Unrecognized event received, hence igonoring");
+        }
+    }
+}
```
Comment (shafreenAnfar) [inline examples/websub-webhook-sample/github_webhook_service.bal:163]: https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

File: examples/websub-webhook-sample/websub-webhook-sample.metatags [hunk 5]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates how a WebSub Subscriber service could be registered as a WebHook Service for GitHub.
+keywords: ballerina, ballerina by example, bbe, websub, subscriber
```
Comment (shafreenAnfar) [inline examples/websub-webhook-sample/websub-webhook-sample.metatags:2]: ditto

File: examples/websub-webhook-sample/github_webhook_service.bal [hunk 6]
```diff
@@ -0,0 +1,163 @@
+// Ballerina WebSub Subscriber service, which could be used as a WebHook Listener for GitHub.
```
Comment (praneesha) [inline examples/websub-webhook-sample/github_webhook_service.bal:1]:
```suggestion
// The Ballerina WebSub Subscriber service, which could be used as a WebHook Listener for GitHub.
```

File: examples/websub-webhook-sample/github_webhook_service.bal [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    string? base_ref;
+    string compare;
+    json[] commits;
+    json? head_commit;
+    Repository repository;
+    json pusher;
+    User sender;
+|};
+
+// Current supported GitHub WebHook event types.
```
Comment (praneesha) [inline examples/websub-webhook-sample/github_webhook_service.bal:134]:
```suggestion
// Currently-supported GitHub WebHook event types.
```

File: examples/websub-webhook-sample/github_webhook_service.bal [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+public type GitHubEvent PingEvent|PushEvent;
+
+// The endpoint to which the webhook service is inbound.
```
Comment (praneesha) [inline examples/websub-webhook-sample/github_webhook_service.bal:137]:
```suggestion
// The endpoint to which the WebHook service is bound.
```

File: examples/websub-webhook-sample/github_webhook_service.bal [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+listener websub:Listener websubEP = new (9090);
+
+// Annotation based configurations specifying the subscription parameters.
```
Comment (praneesha) [inline examples/websub-webhook-sample/github_webhook_service.bal:140]:
```suggestion
// Annotation-based configurations specifying the subscription parameters.
```

File: examples/websub-webhook-sample/github_webhook_service.out [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# To start the service, navigate to the directory that contains the
+# `.bal` file, and use the `bal run` command below.
```
Comment (praneesha) [inline examples/websub-webhook-sample/github_webhook_service.out:2]:
```suggestion
# `.bal` file, and execute the `bal run` command below.
```

File: examples/websub-webhook-sample/websub-webhook-sample.description [hunk 11]
```diff
@@ -0,0 +1,9 @@
+// Ballerina provides the capability to easily introduce subscribers services that are WebSub-compliant.
```
Comment (praneesha) [inline examples/websub-webhook-sample/websub-webhook-sample.description:1]:
```suggestion
// Ballerina provides the capability to easily introduce subscriber services that are WebSub-compliant.
```

File: examples/websub-webhook-sample/websub-webhook-sample.description [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+// Ballerina WebSub subscribers can specify the topic they wish to subscribe to and the hub they wish to subscribe at,
```
Comment (praneesha) [inline examples/websub-webhook-sample/websub-webhook-sample.description:2]:
```suggestion
// Ballerina WebSub subscribers can specify the topic and hub to which they wish to subscribe

```

File: examples/websub-webhook-sample/websub-webhook-sample.description [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+// to receive notifications. If specified, the subscription process will be initiated at the startup. Signature validation
+// is performed by default for subscribers that receive authenticated content. Ballerina WebSub Subscriber
+// Services could thus be registered as webhooks to receive event notifications.
```
Comment (praneesha) [inline examples/websub-webhook-sample/websub-webhook-sample.description:5]:
```suggestion
// Services could thus be registered as WebHooks to receive event notifications.
```

File: examples/websub-webhook-sample/websub-webhook-sample.description [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+//
+// In this example, a WebSub Subscriber service is used to implement GitHub based WebHook Serivice.
```
Comment (praneesha) [inline examples/websub-webhook-sample/websub-webhook-sample.description:7]:
```suggestion
// In this example, a WebSub Subscriber service is used to implement a GitHub-based WebHook service.
```

**Review by praneesha — APPROVED** [review]

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**